internal/provider/resource_tfe_opa_version.go

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceTFEOPAVersion() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEOPAVersionCreate,
+		Read:   resourceTFEOPAVersionRead,
+		Update: resourceTFEOPAVersionUpdate,
+		Delete: resourceTFEOPAVersionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceTFEOPAVersionImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"version": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"sha": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"official": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"beta": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"deprecated": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"deprecated_reason": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  nil,
+			},
+		},
+	}
+}
+
+func resourceTFEOPAVersionCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+
+	opts := tfe.AdminOPAVersionCreateOptions{
+		Version:          *tfe.String(d.Get("version").(string)),
+		URL:              *tfe.String(d.Get("url").(string)),
+		SHA:              *tfe.String(d.Get("sha").(string)),
```

Comment:
Gah, sorry, I did not notice this last time I viewed this. 🥲 

Lauren was right to call this out, _and_ you're right that the create options structs aren't quite the same between OPA and terraform versions. But the upshot is, you do not need to use tfe.String calls in the three fields that take string _values_ rather than string _pointers._ Instead, do:

```suggestion
		Version:          d.Get("version").(string),
		URL:              d.Get("url").(string),
		SHA:              d.Get("sha").(string),
```

`tfe.String()` is a helper to create a string pointer from a given string value... but you don't actually want a pointer, so you can just use the original value instead of creating a pointer and immediately dereferencing it.